src/gcpsample_client/publishing.py

```diff
@@ -75,8 +60,7 @@
 # Now the data has been written, we register the file and the publish
 # is complete. Update the context retention to denote that we're going
 # to save a reference to this entity in our user's history.
-context.retention = context.kPermanent
-final_ref = manager.register(working_ref, video_spec.traitsData(), context)
+final_ref = manager.register(entity_ref, video_spec.traitsData(), context)
```

Comment:
If you're not calling preflight, you can just register to the entity ref you have.